appinventor/appengine/src/com/google/appinventor/client/UIStyleFactory.java

```diff
@@ -0,0 +1,47 @@
+package com.google.appinventor.client;
+
+import com.google.appinventor.client.explorer.folder.ProjectFolder;
+import com.google.appinventor.client.explorer.youngandroid.ProjectList;
+import com.google.appinventor.client.wizards.youngandroid.NewYoungAndroidProjectWizard;
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.json.client.JSONObject;
+import com.google.gwt.uibinder.client.UiBinder;
+import com.google.gwt.uibinder.client.UiTemplate;
+import com.google.gwt.user.client.ui.FlowPanel;
+
+public class UIStyleFactory {
```

Comment:
> It seems to me that it would be better for this to be an `interface` and then have two implementations for classic and modern. The current setup with subclassing in the modern version overrides every single thing about this implementation.

The rationale here was that I wanted the base class to be the default for all additional styles. As we add features, the likelihood increases that styles will lag behind. I even had in mind the possibility that styles could be added that we are not responsible for maintaining. This way, if a method is not implemented in a given style, the system will fall back to Classic.

I also did the Images resource for Neo this way.